src/net/connection.cpp

```diff
@@ -169,23 +169,21 @@ void Connection::handle_resolve(Error error, char type,
 
     if (error) goto finally;
 
-    switch (type) {
-    case DNS_IPv4_A:
-        logger->info("handle_resolve - IPv4");
-        _family = "PF_INET";
-        break;
-    case DNS_IPv6_AAAA:
-        logger->info("handle_resolve - IPv6");
-        _family = "PF_INET6";
-        break;
-    default:
-        abort();
-    }
-
-    for (auto &address : results) {
+    for (auto answer: answers) {
+        std::string address;
+        if (answer.type == dns::QueryTypeId::A) {
+            address = answer.ipv4;
+            _family = "PF_INET";
+        } else if (answer.type == dns::QueryTypeId::AAAA) {
+            address = answer.ipv6;
+            _family = "PF_INET6";
+        } else {
+             abort();
```

Comment:
Note to self: here it would be better to throw an exception (a kind of assertion that could be checked in the regress tests) rather than calling `abort()` (which instead is something that you cannot intercept).